dev/reference/tools/data-migration/upgrade.md

```diff
@@ -9,195 +9,131 @@ aliases: ['/docs/dev/how-to/upgrade/data-migration/']
 Git Branch: dm-master
-UTC Build Time: 2018-12-19 04:58:46
+UTC Build Time: 2019-07-24 02:26:08
 Go Version: go version go1.11.2 linux/amd64
 ```
 
 ### Main changes
 
-Before this version, `mysql-instances` in the task configuration file (`task.yaml`) contains the following information:
-
-- `config`: the upstream MySQL address, user name and password.
-- `instance-id`: identifies an upstream MySQL.
-
-Starting from this version, the above two kinds of information are removed and the following information is added:
-
-- `source_id`: exists in `inventory.ini`, used to identify an upstream MySQL instance or a master-slave replication group.
-- `source-id`: exists in `mysql-instances` of the task configuration file. Its value corresponds to `source_id` of `inventory.ini`.
-
-> **Note:**
->
-> To guarantee that the checkpoint information of the existing task stored in the downstream database can still be used, the value of `source_id`or `source-id` should be consistent with that of `instance-id` of the corresponding DM-worker.
-
-### Upgrade operation example
-
-1. Modify the `inventory.ini` configuration information.
-
-    Set the corresponding `source_id` for all DM-worker instances.
-
-    For example, modify
-
-    ```ini
-    dm-worker1 ansible_host=172.16.10.72 server_id=101 mysql_host=172.16.10.72 mysql_user=root mysql_password='VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU=' mysql_port=3306
-    ```
-
-    to
-
-    ```ini
-    dm-worker1 ansible_host=172.16.10.72 source_id="mysql-replica-01" server_id=101 mysql_host=172.16.10.72 mysql_user=root mysql_password='VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU=' mysql_port=3306
-    ```
-
-2. Use DM-Ansible to perform a rolling update on DM.
-
-3. Modify the task configuration file (`task.yaml`).
-
-    Remove the `config` and `instance-id` configuration items and add the `source-id` configuration item (corresponding to `source_id` in `inventory.ini`).
-
-    For example, modify
+Starting from this release, TiDB DM checks all configurations rigorously. Unrecognized configuration triggers an error. This is to ensure that users always know exactly what the configuration is.
 
-    ```yaml
-    config:
-          host: "192.168.199.118"
-          port: 4306
-          user: "root"
-          password: "1234"
-    instance-id: "instance118-4306" # It is unique. It is used as the ID when storing the checkpoint, configuration and other information.
-    ```
+### Upgrade notes
 
-    to
+Before starting the DM-master or DM-worker, ensure that the obsolete configuration information has been deleted and there are no redundant configuration items.
 
-    ```yaml
-    source-id: "instance118-4306" # It should be consistent with the original `source_id` value when the checkpoint of the original task needs to be reused.
-    ```
+Otherwise, the starting might fail. In this situation, you can delete the redundant configuration based on the failure information. These are two possible redundant configurations:
```

Comment:
`deprecated` differs from `redundant`.

```suggestion
Otherwise, the starting might fail. In this situation, you can delete the deprecated configuration based on the failure information. These are two possible deprecated configurations:
```